src/couch_replicator/src/couch_replicator_scheduler_job.erl

```diff
@@ -652,16 +652,39 @@ changes_manager_loop_open(Parent, ChangesQueue, BatchSize, Ts) ->
         case couch_work_queue:dequeue(ChangesQueue, BatchSize) of
         closed ->
             From ! {closed, self()};
-        {ok, Changes} ->
-            #doc_info{high_seq = Seq} = lists:last(Changes),
-            ReportSeq = {Ts, Seq},
-            ok = gen_server:cast(Parent, {report_seq, ReportSeq}),
+        {ok, ChangesOrLastSeqs} ->
+            {LastSeqs, Changes} = lists:partition(
+                fun({last_seq, _Seq}) ->
+                    true;
+                (#doc_info{}) ->
+                    false
+            end, ChangesOrLastSeqs),
+            ReportSeq = maybe_report_in_progress(Parent, Changes, From, Ts),
+            ok = maybe_report_last_sequence(Parent, LastSeqs, Ts),
             From ! {changes, self(), Changes, ReportSeq}
         end,
         changes_manager_loop_open(Parent, ChangesQueue, BatchSize, Ts + 1)
     end.
 
 
+maybe_report_last_sequence(_Parent, [] = _LastSeqs, _Ts) ->
+    ok;
+maybe_report_last_sequence(Parent, LastSeqs, Ts) ->
+    {last_seq, Seq} = lists:last(LastSeqs),
+    Msg = {report_seq_done, {Ts, Seq}, couch_replicator_stats:new()},
+    ok = gen_server:call(Parent, Msg, infinity).
+
+
+maybe_report_in_progress(_Parent, [] = _Changes, _From, _Ts) ->
+    nil;
```

Comment:
ReportSeq could end up being nil here, will that not break a worker? Or are you relying on the empty clause match?